SecurityPkg/VariableAuthenticated/SecureBootConfigDxe/SecureBootConfigImpl.c

```diff
@@ -4651,8 +4705,51 @@ SecureBootCallback (
       // When entering SecureBoot OPTION Form
       // always close opened file & free resource
       //
-      if ((QuestionId == KEY_SECURE_BOOT_PK_OPTION) ||
-          (QuestionId == KEY_SECURE_BOOT_KEK_OPTION) ||
+      if ((QuestionId == KEY_SECURE_BOOT_PK_OPTION)) {
+        CloseEnrolledFile (Private->FileContext);
+
+        GetVariable2 (EFI_PLATFORM_KEY_NAME, &gEfiGlobalVariableGuid, (VOID **)&PkList, NULL);
+        if (PkList != NULL) {
+          // Extract the PK certificate from the list
+          PkCert = (EFI_SIGNATURE_DATA *)((UINT8 *)PkList + sizeof (EFI_SIGNATURE_LIST) + PkList->SignatureHeaderSize);
+
+          // Get the required buffer size
+          UINTN cnSize = 0;
+          X509GetCommonName (
+            (UINT8*)PkCert->SignatureData,
+            (UINTN)PkList->SignatureSize,
+            NULL,
+            &cnSize
+            );
+
+          PkCN8 = AllocateZeroPool (cnSize * sizeof(CHAR8));
+          if (PkCN8 == NULL) {
+            Status = EFI_OUT_OF_RESOURCES;
+            goto EXIT;
+          }
+          PkCN16 = AllocateZeroPool (cnSize * sizeof(CHAR16));
+          if (PkCN16 == NULL) {
+            Status = EFI_OUT_OF_RESOURCES;
+            goto EXIT;
+          }
+
+          RStatus = X509GetCommonName (
+            (UINT8*)PkCert->SignatureData,
+            (UINTN)PkList->SignatureSize,
+            PkCN8,
+            &cnSize
+            );
+
+          if (!EFI_ERROR(RStatus)) {
+            AsciiStrToUnicodeStrS (PkCN8, PkCN16, cnSize);
+            HiiSetString (Private->HiiHandle, STRING_TOKEN (STR_PK_NAME), PkCN16, NULL);
+          } else {
+            HiiSetString (Private->HiiHandle, STRING_TOKEN (STR_PK_NAME), L"*NO COMMON NAME*", NULL);

```

Comment:
```suggestion
            HiiSetString (Private->HiiHandle, STRING_TOKEN (STR_PK_NAME), UNKNOWN_CERT, NULL);
```

Shouldn't we use `UNKNOWN_CERT` for consistency with how other certs are handled?

Reply:
Switched to `UNKNOWN_CERT` [here](https://github.com/Dasharo/edk2/compare/f2baf9f60b923e548f716217d022975fad0258d5..b60143bf81544810caea7ea1c67add0921bd80ff#diff-49fdb815b38763cce7b30e2aea02c17111bc51206950ff1d21e0073ebd0cb1d3R4747)